website/source/docs/providers/kubernetes/r/secret.html.markdown

```diff
@@ -0,0 +1,64 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_secret"
+sidebar_current: "docs-kubernetes-resource-secret"
+description: |-
+  The resource provides mechanisms to inject containers with sensitive information while keeping containers agnostic of Kubernetes.
+---
+
+# kubernetes_secret
+
+The resource provides mechanisms to inject containers with sensitive information, such as passwords, while keeping containers agnostic of Kubernetes.
+Secrets can be used to store sensitive information either as individual properties or coarse-grained entries like entire files or JSON blobs.
+
+## Example Usage
+
+```
+resource "kubernetes_secret" "example" {
+  metadata {
+    name = "basic-auth"
+  }
+
+  data {
+    username = "admin"
+    password = "P4ssw0rd"
+  }
+
+  type = "kubernetes.io/basic-auth"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `data` - (Optional) A map of the secret data.
+* `metadata` - (Required) Standard secret's metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+* `type` - (Optional) The secret type. Defaults to `Opaque`. More info: https://godoc.org/k8s.io/kubernetes/pkg/api#SecretType
```

Comment:
It's shame these types aren't documented in the official docs, but how do you feel about linking to this document instead? https://github.com/kubernetes/community/blob/master/contributors/design-proposals/secrets.md#proposed-design - I think it's a bit more condensed and to-the-point compared to the godoc link.